### PR TITLE
fix tests referencing top level of sharepointInfo for listInfo

### PR DESCRIPTION
**Pull request by HiteshRepo**
fix tests referencing top level of SharepointInfo for list details instead of nested ListInfo.
also disables a test that requires list backup to be enabled in CLI level.

#### Does this PR need a docs update or release note?
- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix

#### Issue(s)
https://github.com/alcionai/corso/actions/runs/7312375792/job/19922878417#step:6:666

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/site/collection_test.go [hunk 1]
```diff
@@ -233,9 +237,15 @@ func (suite *SharePointCollectionSuite) TestCollection_Items() {
...
 
-			assert.NotNil(t, info)
-			assert.NotNil(t, info.SharePoint)
-			assert.Equal(t, test.itemName, info.SharePoint.ItemName)
+			require.NotNil(t, info)
+			require.NotNil(t, info.SharePoint)
+
+			if test.scope.Category().String() == selectors.SharePointList.String() {
+				require.NotNil(t, info.SharePoint.List)
+				assert.Equal(t, test.itemName, info.SharePoint.List.Name)
```
Comment (meain) [inline src/internal/m365/collection/site/collection_test.go:245]: Didn't know we were not using the original `ItemName`.

File: src/cli/backup/sharepoint_e2e_test.go [hunk 2]
```diff
@@ -341,7 +342,9 @@ func runSharepointDetailsCmdTest(suite *PreparedBackupSharepointE2ESuite, catego
 	foundList := 0
 
 	for _, ent := range deets.Entries {
-		if ent.SharePoint != nil && ent.SharePoint.ItemName != "" {
+		if ent.SharePoint != nil &&
```
Comment (meain) [inline src/cli/backup/sharepoint_e2e_test.go:345]: Looks like we ended up shipping it for items that are not lists.

**Review by meain — CHANGES_REQUESTED** [review]

File: src/internal/m365/collection/site/collection_test.go (hunk 1 above)
Comment (HiteshRepo) [reply src/internal/m365/collection/site/collection_test.go:245]: @meain  for lists only populating 'name' of [ListInfo](https://github.com/alcionai/corso/blob/43ca43910fdb241a8ae5d917a6500bcea274cab4/src/pkg/services/m365/api/lists.go#L623) and not the itemName in SharepointInfo

Comment (HiteshRepo) [reply src/internal/m365/collection/site/collection_test.go:245]: yeah. Let me make the change to populate `itemName` name instead.

Comment (meain) [reply src/internal/m365/collection/site/collection_test.go:245]: No need to change it if this has already been accepted to be how we do it. I was just thinking out loud.

File: src/cli/backup/sharepoint_e2e_test.go (hunk 2 above)
Comment (HiteshRepo) [reply src/cli/backup/sharepoint_e2e_test.go]: @meain  attempted to make it generic instead of renaming the function

Comment (meain) [reply src/cli/backup/sharepoint_e2e_test.go]: Hmm, not sure if we are on the same page here. The function was previously checking for details of SharePoint drive items and hence `ent.SharePoint.ItemName` check, but now that case is never checked as `ent.SharePoint.List` will be `nil` for drive items.

**Review by pandeyabs — APPROVED** [review]

File: src/cli/backup/sharepoint_e2e_test.go (hunk 2 above)
Comment (HiteshRepo) [reply src/cli/backup/sharepoint_e2e_test.go]: @meain please take a look at the current change. Now it caters to both sharepoint list and drive items.

**Pull request merged**